storage/conformance_test.go

```diff
@@ -16,21 +16,337 @@ package storage
+			return fmt.Errorf("creating bucket: %v", err)
+		}
+		attrs, err := bkt.Attrs(ctx)
+		if err != nil {
+			return fmt.Errorf("getting bucket attrs: %v", err)
+		}
+		fs.bucket = attrs
+	case storage_v1_tests.Resource_OBJECT:
+		// Assumes bucket has been populated first.
+		obj := c.Bucket(fs.bucket.Name).Object(objectIDs.New())
+		w := obj.NewWriter(ctx)
+		if _, err := w.Write([]byte("abcdef")); err != nil {
+			return fmt.Errorf("writing object: %v", err)
+		}
+		if err := w.Close(); err != nil {
+			return fmt.Errorf("closing object: %v", err)
+		}
+		attrs, err := obj.Attrs(ctx)
+		if err != nil {
+			return fmt.Errorf("getting object attrs: %v", err)
+		}
+		fs.object = attrs
+	case storage_v1_tests.Resource_NOTIFICATION:
+		// Assumes bucket has been populated first.
+		n, err := c.Bucket(fs.bucket.Name).AddNotification(ctx, &Notification{
+			TopicProjectID: projectID,
+			TopicID:        notificationIDs.New(),
+			PayloadFormat:  JSONPayload,
+		})
+		if err != nil {
+			return fmt.Errorf("adding notification: %v", err)
+		}
+		fs.notification = n
+	case storage_v1_tests.Resource_HMAC_KEY:
+		key, err := c.CreateHMACKey(ctx, projectID, serviceAccountEmail)
+		if err != nil {
+			return fmt.Errorf("creating HMAC key: %v", err)
+		}
+		fs.hmacKey = key
+	}
+	return nil
+}
+
+type retryFunc func(ctx context.Context, c *Client, fs *resources, preconditions bool) error
+
+// Methods to retry. This is a map whose keys are a string describing a standard
+// API call (e.g. storage.objects.get) and values are a list of functions which
+// wrap library methods that implement these calls. There may be multiple values
+// because multiple library methods may use the same call (e.g. get could be a
+// read or just a metadata get).
+var methods = map[string][]retryFunc{
+	"storage.buckets.get": {
+		func(ctx context.Context, c *Client, fs *resources, _ bool) error {
+			_, err := c.Bucket(fs.bucket.Name).Attrs(ctx)
+			return err
+		},
+	},
+}
+
+func TestRetryConformance(t *testing.T) {
+	host := os.Getenv("STORAGE_EMULATOR_HOST")
+	if host == "" {
+		t.Skip("This test must use the testbench emulator; set STORAGE_EMULATOR_HOST to run.")
+	}
+
+	ctx := context.Background()
+
+	// Create non-wrapped client to use for setup steps.
+	client, err := NewClient(ctx)
+	if err != nil {
+		t.Fatalf("storage.NewClient: %v", err)
+	}
+
+	_, _, testFiles := parseFiles(t)
+
+	for _, testFile := range testFiles {
+		for _, retryTest := range testFile.RetryTests {
+			for _, instructions := range retryTest.Cases {
+				for _, method := range retryTest.Methods {
+					if len(methods[method.Name]) == 0 {
+						t.Logf("No tests for operation %v", method.Name)
+					}
+					for i, fn := range methods[method.Name] {
+						testName := fmt.Sprintf("%v-%v-%v-%v", retryTest.Id, instructions.Instructions, method.Name, i)
+						t.Run(testName, func(t *testing.T) {
+
+							// Create the retry subtest
+							subtest := &retrySubtest{T: t, name: testName}
```

Comment:
"subtest" seems like the wrong word here-- this creates/manages a retry test in the emulator, correct? Any thoughts on a better name?